operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/ConfiguredController.java

```diff
@@ -0,0 +1,169 @@
+package io.javaoperatorsdk.operator.processing;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Objects;
+
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.operator.CustomResourceUtils;
+import io.javaoperatorsdk.operator.MissingCRDException;
+import io.javaoperatorsdk.operator.OperatorException;
+import io.javaoperatorsdk.operator.api.Context;
+import io.javaoperatorsdk.operator.api.DeleteControl;
+import io.javaoperatorsdk.operator.api.ResourceController;
+import io.javaoperatorsdk.operator.api.UpdateControl;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+import io.javaoperatorsdk.operator.processing.event.DefaultEventSourceManager;
+import io.javaoperatorsdk.operator.processing.event.EventSourceManager;
+
+public class ConfiguredController<R extends CustomResource<?, ?>> implements ResourceController<R>,
```

Comment:
I don't think this should implement ResourceController, couldn't we just get the controller when it's needed instead of delegating and implementing this interface. I would rather see this class which just couples these objects. Implementing this interface does not seems right.

In other words this principle loosly applies (in regard of interfaces):
https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
We're using composition here and to me, it makes sense to have that class implement `ResourceController`. We're not inheriting anything, just stating that this class behaves for all intents and purposes like a `ResourceController`. What are the disadvantages of having the class extend the interface? Implementing the interface allows to pass the object around anywhere a `ResourceController` is used currently which simplifies the current code, imo. The whole idea of this class is that it **is** a `ResourceController` that also manages its configuration (instead of asking a `ConfigurationService` for it).

Reply:
This is probably more a philosphical question, so also a point of view how we are looking at this class. So if we take a look on it as a "proxy" class then yes you are right there is no harm to implement it. If we take a look on this as a wrapper that just collects some related classes then it is a smell. 
So it's all about how we express notions. But here actually you are probably right that we can look on this as a "proxy" class.